GraknClient.java

```diff
@@ -38,29 +41,74 @@ static GraknClient core(String address) {
         return new ClientRPC(address);
     }
 
-    static GraknClient cluster(String... addresses) {
+    static GraknClient.Cluster cluster(String... addresses) {
         return new ClientClusterRPC(addresses);
     }
 
-    GraknClient.Session session(String database, GraknClient.Session.Type type);
+    Session session(String database, GraknClient.Session.Type type);
 
-    GraknClient.Session session(String database, GraknClient.Session.Type type, GraknOptions options);
+    Session session(String database, GraknClient.Session.Type type, GraknOptions options);
 
-    GraknClient.DatabaseManager databases();
+    DatabaseManager databases();
 
     boolean isOpen();
 
     void close();
 
+    interface Cluster extends GraknClient {
+
+        @Override
+        DatabaseManager.Cluster databases();
+    }
+
     interface DatabaseManager {
 
         boolean contains(String name);
 
+        // TODO: Return type should be 'Database' but right now that would require 2 server calls in Cluster
         void create(String name);
 
-        void delete(String name);
+        Database get(String name);
+
+        List<? extends Database> all();
```

Comment:
Can't use `List<Database>` because `List<Database.Cluster>` is a weaker type constraint than `List<Database>`